src/lab14.py

```diff
@@ -668,9 +668,10 @@ def intersects(self, x, y):
         return False
 
     def hit_test(self, x, y):
-        for node in tree_to_list(self, []):
-            if node.bounds.intersects(x, y):
-                return node
+        nodes = [node for node in tree_to_list(self, [])
```

Comment:
The old code was wrong because you need to hit test from the bottom of the tree up.

Reply:
Oh yeah, you are right.